### ISSUE TITLE
Feedback for spectral search

### ISSUE THREAD

**Issue by tomas-pluskal** [mzmine/mzmine2#728]
Hi @robinschmid 

I have some feedback for the spectral search module after discussing with people who use it:

- The precursor m/z tolerance should be an optional parameter (same as RT tolerance). If unchecked, the precursor mass should be ignored. This is useful for searching for similar fragments coming from different precursors.
- For the weights in the Similarity parameters, wouldn't it be useful to add one more option that completely ignores the intensities (in other words, compare spectra entirely based on presence or absence of certain fragments, disregarding their intensities)
- Currently the GNPS export module exports an MGF file, but the spectral search module expects a JSON file. There seems to be no obvious way to export a spectral library from an MZmine peak list that could be then used for spectral search (i.e., in another MZmine project). What do you think about that?
- When we tried use an MGF file for the spectral library, the module finished immediately with no error message. Perhaps there should be some sanity check for the input file format and the module should give some feedback or warning if no suitable spectra are found.

**Comment by robinschmid** [comment]
Have to do this later. 
I am also linking this out to @Annexhc

**Comment by robinschmid** [comment]
- include ADAP data structure for search (correlated MS1 spectra)

**Comment by asmirn1** [comment]
Just a thought: MZmine currently have about 6 database search modules 

- Local spectra database search
- Precursor search in local spectral DB
- Custom database search
- Online database search
- Lipid search
- NIST MS Search

It seems that at least four of them (Local spectra database search, Precursor search in local spectral DB, Custom database search, and Lipid search) have overlapping functionality. Maybe, it would make sense to combine them into one or two modules?

Overall, the __Identification__ menu seems overcrowded with 15 modules. It would me helpful to split it into several categories.

**Comment by tomas-pluskal** [comment]
@asmirn1 Good point! Restructuring the MZmine menu is definitely on my task list!

**Comment by robinschmid** [comment]
Precursor search in local DB and Local spectra database search can definitely be merged into one. We basically need to add a boolean **tag putative precursors** for this.

**Comment by Annexhc** [comment]
@asmirn1 I agree, the menu is quite crowded. We could add e.g. sub menus. I dont see to much potential combining most of the modules, except Local spectra database search and Precursor search in local spectral DB. E.g. custom DB search is simple MS1 and RT matching and Lipid Search includes rule based MS2 identifications. 
We should definitely harmonize the module names, e.g. a custom database is also a local database or the other way around.

**Comment by asmirn1** [comment]
I think it's a good idea to rename modules in a more consistent way. Here are some idea for renaming:

- If a module uses local MSP, CSV, etc. files to store the spectral library, then it should have __local database (or library) search__ in its name,

- If a module works with only MS1 or MS2 spectra, then it should have __MS1 only__ or __MS2 only__ in its name,

- Modules like _Fragment search_, _Adduct search_, _Lipid Search_ are quite different in terms of underlying algorithms and their results. I find it confusing that they are all called "search".

**Comment by tomas-pluskal** [comment]
I would say this kind of refactoring might be good to do at the point where we switch the version number from 2.xx to 3.0. Because then users won't be surprised that the menu has changed and modules have different names. For remaining 2.xx versions, I would probably leave the menu as it is..

I am planning to setup the repository for mzmine3 soon.
